politeiawww/politeiawww.go

```diff
@@ -858,6 +865,60 @@ func (p *politeiawww) handleWebsocketWrite(wc *wsContext) {
 	}
 }
 
+// updateBestBlock updates the cached best block.
+func (p *politeiawww) updateBestBlock(bestBlock uint64) {
+	p.bbMtx.Lock()
+	defer p.bbMtx.Unlock()
+	p.bestBlock = bestBlock
+}
+
+// getBestBlock returns the cached best block if there is an active websocket
+// connection to dcrdata. Otherwise it requests the best block from politeiad
+// using the the decred plugin best block command.
+func (p *politeiawww) getBestBlock() (uint64, error) {
+	if !p.wsDcrdata.isSubscribed(newBlockSub) {
+		return p.getBestBlockDecredPlugin()
+	}
+
+	p.bbMtx.RLock()
+	defer p.bbMtx.RUnlock()
+	return p.bestBlock, nil
+}
+
+// setupPiDcrdataWSSubs subscribes and listens to websocket messages from
+// dcrdata that are needed for pi.
+func (p *politeiawww) setupPiDcrdataWSSubs() error {
+	err := p.wsDcrdata.subToNewBlock()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			msg, ok := <-p.wsDcrdata.client.Receive()
+			if !ok {
+				log.Errorf("Error receiving msg from dcrdata, msg: %v", msg)
+				p.wsDcrdata.reconnect()
+				continue
+			}
+
+			switch m := msg.Message.(type) {
+			case *exptypes.WebsocketBlock:
+				log.Debugf("wsDcrdata message WebsocketBlock(height=%s)",
+					m.Block.Height)
+				p.updateBestBlock(uint64(m.Block.Height))
+			case *pstypes.HangUp:
+				log.Infof("Dcrdata has hung up. Will reconnect...")
+				p.wsDcrdata.reconnect()
+			default:
+				log.Errorf("Message of type %v unhandled. %v", msg.EventId, m)
```

Comment:
Make this explicit so that if we add future websocket connections we'll know what this is referring to.
```suggestion
				log.Errorf("wsDcrdata message of type %v unhandled. %v",
					msg.EventId, m)
```